### PR TITLE
All files generated by schemazen now has a comment in the beginning of the file

### PR DESCRIPTION
**Pull request by douglas-wong-zocdoc**
@Zocdoc/ci-devs @colin-hanson-zocdoc 
Please review.

### CODE REVIEW

**Comment by marcio-santos-zocdoc** [comment]
@douglas-wong-zocdoc, why did you close this PR? Do you still need someone to review it?

File: model/Models/Database.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 using System;
+using System.CodeDom;
```
Comment (marcio-santos-zocdoc) [inline model/Models/Database.cs]: Do we need this namespace?

File: model/Models/Database.cs [hunk 2]
```diff
@@ -59,10 +60,12 @@ public Database(string name, IList<string> filteredTypes = null)
 		public const string SqlWhitespaceOrCommentRegex = @"(?>(?:\s+|--.*?(?:\r|\n)|/\*.*?\*/))";
 		public const string SqlEnclosedIdentifierRegex = @"\[.+?\]";
 		public const string SqlQuotedIdentifierRegex = "\".+?\"";
-
 		public const string SqlRegularIdentifierRegex = @"(?!\d)[\w@$#]+";
 			// see rules for regular identifiers here https://msdn.microsoft.com/en-us/library/ms175874.aspx
 
+	    public const string AutoGenerateComment =
+	        "/* This file is automatically generated with SchemaZen, please do not modify. */\r\n";
```
Comment (marcio-santos-zocdoc) [inline model/Models/Database.cs]: The comment could be something like this:

```
//----------------------------------------------------------------------------
// <auto-generated>
//     This file was generated by schemazen.
//     Date: ___
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//----------------------------------------------------------------------------
```

File: test/CommentTester.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+namespace SchemaZen.Tests
+{
+    [TestFixture]
+    class CommentTester {
+        private Database _db;
+        private string _comment;
+
+        [SetUp]
+        public void SetUp() {
+            _db = new Database("TestScriptTableType");
```
Comment (marcio-santos-zocdoc) [inline test/CommentTester.cs]: If you create a database in a test, you need to destroy it in the end.

File: test/CommentTester.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+        [Test]
+        public void TestFilesContainComment()
+        {
+            _db.Dir = _db.Name;
+            _db.Load();
+            _db.ScriptToDir();
+
+            Assert.IsTrue(validateFirstLineIncludesComment(_db.Name +
+                "\\table_types\\" + TestStrings.TableTypeFileName, _comment));
```
Comment (marcio-santos-zocdoc) [inline test/CommentTester.cs]: You can check the entire message instead of just a word.

File: test/CommentTester.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            Assert.IsTrue(validateFirstLineIncludesComment(_db.Name +
+                "\\foreign_keys\\" + TestStrings.TestForeignKeyFileName, _comment));
+            Assert.IsTrue(validateFirstLineIncludesComment(_db.Name +
+                "\\functions\\" + TestStrings.TestFunctionFileName, _comment));
+            Assert.IsTrue(validateFirstLineIncludesComment(_db.Name +
```
Comment (marcio-santos-zocdoc) [inline test/CommentTester.cs]: You can use test cases instead of having multiple assertions: http://www.nunit.org/index.php?p=testCase&r=2.5

**Comment by douglas-wong-zocdoc** [comment]
@marcio-santos-zocdoc Closed the PR because I realized I didn't do a teardown, but will be fixing this along with the other issues, thanks!

**Comment by marcio-santos-zocdoc** [comment]
Great,

Don't be afraid to delete PRs that you don't need anymore.

On Jul 5, 2016, at 15:48, Douglas Wong <notifications@github.com<mailto:notifications@github.com>> wrote:

@marcio-santos-zocdochttps://github.com/marcio-santos-zocdoc Closed the PR because I realized I didn't do a teardown, but will be fixing this along with the other issues, thanks!

##

**Comment by douglas-wong-zocdoc** [comment]
Will do!

File: model/Models/Database.cs (hunk 1 above)
Comment (douglas-wong-zocdoc) [inline model/Models/Database.cs]: Not sure how this got in, fixed. In the future will look more closely at my diffs.

File: model/Models/Database.cs (hunk 2 above)
Comment (douglas-wong-zocdoc) [inline model/Models/Database.cs]: Thanks, this is now fixed.

File: test/CommentTester.cs (hunk 5 above)
Comment (douglas-wong-zocdoc) [inline test/CommentTester.cs]: Thanks, this has been fixed.

File: test/CommentTester.cs (hunk 4 above)
Comment (douglas-wong-zocdoc) [inline test/CommentTester.cs]: My thinking was this makes sure the comment is on top of the file, would that be unnecessary?

File: model/Models/Database.cs (hunk 1 above)
Comment (douglas-wong-zocdoc) [inline model/Models/Database.cs]: Fixed.

File: test/CommentTester.cs (hunk 3 above)
Comment (douglas-wong-zocdoc) [inline test/CommentTester.cs]: Thanks, this has been fixed.

**Comment by douglas-wong-zocdoc** [comment]
@colin-hanson-zocdoc @marcio-santos-zocdoc All issues fixed. Also reformatted some of the code to conform to Schemazen's style.

**Comment by douglas-wong-zocdoc** [comment]
woops changing some variable names now.

**Comment by douglas-wong-zocdoc** [comment]
now good to go.

**Comment by marcio-santos-zocdoc** [comment]
:+1: :shipit:

**Comment by marcio-santos-zocdoc** [comment]
@douglas-wong-zocdoc after merging to schemazen, we need to:
- update Clean DB to use the new version
- merge changes to original schemazen repo

**Pull request merged**